### PR TITLE
refactor browsing agent response parse

### PR DESCRIPTION
**Pull request by yufansong**
As title.

### CODE REVIEW

File: agenthub/browsing_agent/browsing_agent.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import ast
 import os
 
 from browsergym.core.action.highlevel import HighLevelActionSet
 from browsergym.utils.obs import flatten_axtree_to_str
 
+from agenthub.browsing_agent.resposne_parser import BrowsingResponseParser
```
Comment (SmartManoj) [inline agenthub/browsing_agent/browsing_agent.py:6]:
```suggestion
from agenthub.browsing_agent.response_parser import BrowsingResponseParser
```

File: agenthub/browsing_agent/browsing_agent.py [hunk 2]
```diff
@@ -42,6 +42,7 @@ class BrowsingAgent(Agent):
 
     sandbox_plugins: list[PluginRequirement] = []
     runtime_tools: list[RuntimeTool] = [RuntimeTool.BROWSER]
+    resposne_parser = BrowsingResponseParser()
```
Comment (SmartManoj) [inline agenthub/browsing_agent/browsing_agent.py:45]:
```suggestion
    response_parser = BrowsingResponseParser()
```

File: agenthub/browsing_agent/browsing_agent.py [hunk 3]
```diff
@@ -216,19 +192,14 @@ def step(self, state: State) -> Action:
...
         )
         self.log_cost(response)
-        action_resp = response['choices'][0]['message']['content'].strip()
-        if not action_resp.endswith('```'):
-            action_resp = action_resp + ')```'
-
-        logger.info(prompt)
-        logger.info(action_resp)
-        return self.parse_response(action_resp)
+        return self.resposne_parser.parse(response)
```
Comment (SmartManoj) [inline agenthub/browsing_agent/browsing_agent.py:202]:
```suggestion
        return self.response_parser.parse(response)
```

**Review by SmartManoj — COMMENTED** [review]
Fixed typos

File: agenthub/browsing_agent/response_parser.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    Action,
+    BrowseInteractiveAction,
+)
+
+
+class BrowsingResponseParser(ResponseParser):
+    def __init__(
+        self,
+    ):
+        # Need pay attention to the item order in self.action_parsers
```
Comment (li-boxuan) [inline agenthub/browsing_agent/response_parser.py:15]:
```suggestion
        # Need to pay attention to the item order in self.action_parsers
```

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**